packages/dashboard/src/components/robots/robot-summary.tsx

```diff
@@ -0,0 +1,253 @@
+  }
+};
+
+const LinearProgressWithLabel = (props: LinearProgressProps & { value: number }) => {
+  return (
+    <Box sx={{ display: 'flex', alignItems: 'center' }}>
+      <Box sx={{ width: '100%', mr: 1 }}>
+        <LinearProgress variant="determinate" {...props} />
+      </Box>
+      <Box sx={{ minWidth: 35 }}>
+        <Typography variant="body2" color="text.secondary">{`${Math.round(
+          props.value,
+        )}%`}</Typography>
+      </Box>
+    </Box>
+  );
+};
+
+interface RobotSummaryProps {
+  onClose: () => void;
+  robot: RobotTableData;
+}
+
+export const RobotSummary = React.memo(({ onClose, robot }: RobotSummaryProps) => {
+  const classes = useStyles();
+  const rmf = React.useContext(RmfAppContext);
+
+  const [isOpen, setIsOpen] = React.useState(true);
+  const [robotState, setRobotState] = React.useState<RobotState | null>(null);
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+  const [openTaskDetailsLogs, setOpenTaskDetailsLogs] = React.useState(false);
+  const [location, setLocation] = React.useState('');
+  const [destination, setDestination] = React.useState('');
+
+  React.useEffect(() => {
+    if (!rmf) {
+      return;
+    }
+    const sub = rmf
+      .getFleetStateObs(robot.fleet)
+      .pipe(
+        mergeMap((fleetState) => {
+          const robotState = fleetState?.robots?.[robot.name];
+          const taskObs = robotState?.task_id ? rmf.getTaskStateObs(robotState.task_id) : of(null);
+          return robotState ? combineLatest([of(robotState), taskObs]) : EMPTY;
+        }),
+      )
+      .subscribe(([robotState, taskState]) => {
+        setRobotState(robotState);
+        setTaskState(taskState);
+      });
+    return () => sub.unsubscribe();
+  }, [rmf, robot.fleet, robot.name]);
+
+  const taskProgress = React.useMemo(() => {
+    if (
+      !taskState ||
+      !taskState.estimate_millis ||
+      !taskState.unix_millis_start_time ||
+      !taskState.unix_millis_finish_time
+    ) {
+      console.log(`Can't calculate task progress`);
+      return undefined;
+    }
+
+    return Math.min(
+      1.0 -
+        taskState.estimate_millis /
+          (taskState.unix_millis_finish_time - taskState.unix_millis_start_time),
+      1,
+    );
+  }, [taskState]);
+
+  React.useEffect(() => {
+    if (!taskState || !taskState.phases || !taskState.active) {
+      setLocation('Failed to retrieve current location');
+      setDestination('Failed to retrieve robot destination');
+      return;
+    }
+
+    const message = Object.values(taskState.phases)[taskState.active - 1]?.detail;
+
+    if (message) {
+      const regex = /\[place:(.*?)\]/g;
+
+      let match;
+      const waypoints = [];
+
+      // Iterate over all matches found by the regular expression
+      while ((match = regex.exec(message.toString()))) {
+        waypoints.push(match[1]);
+      }
+      setLocation(waypoints[0]);
+      setDestination(waypoints[1]);
+    }
```

Comment:
I think we are missing the else case here, where we really fail to retrieve the navigation start and destination. In the else case we should probably still display these fields as `-`, but use console errors to notify that we are unable to retrieve the information needed.

Reply:
Done! 5c7929f75e267afa7e53069cde2af45337555aea